crates/notion-core/src/distro/mod.rs

```diff
@@ -33,9 +34,35 @@ impl<V> Fetched<V> {
     }
 }
 
-pub trait Distro: Sized {
-    type VersionDetails;
+/// Abstraction to contain info about Distro versions.
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub enum DistroVersion {
```

Comment:
Something feels suspicious about this being an `enum` instead of (1) a trait (2) a separate type for each kind of distro and (3) an associated type in the `Distro` trait. By using an `enum`, we aren't using the type system to guarantee at compile time that e.g. a `NodeDistro` will only ever produce a `DistroVersion::Node`. I think maybe the runtime checks (`if let`) in `FetchResolve<NodeDistro>::fetch()` and `FetchResolve<YarnDistro>::fetch()` are a consequence of this.

Reply:
Ah, I see @charlespierce made the same point. Probably easier to talk through this synchronously…

Reply:
So in that thread, you were unsure how to make it generic since you need to call `inventory.fetch()`. One thought is that you could make each collection crate-visible, so the dynamic version is still `inventory.fetch()`, but when you're statically dealing with Node you can call `inventory.node.fetch()`, when you're statically dealing with Yarn you can call `inventory.yarn.fetch()`, etc. Would something like that work?